### ISSUE TITLE
Exception for optional array fields

### ISSUE THREAD

**Issue by themerk** [typesense/typesense-instantsearch-adapter#100]
I'm working with documents that have lots of optional fields, which are often missing. This seems to cause problems on string array fields.

In this particular case, I have a field which is:
- Optional, typed `string[]`
- Present in the `query_by` parameter but NOT the `include_fields` parameter (since I don't need it in the result)
- Not present in one of the documents returned in the results

That causes an exception on this line:
`Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'forEach')`
https://github.com/typesense/typesense-instantsearch-adapter/blob/f8e59eeb8df1d943e3b5a8f99065d3aa6ecd79ae/src/SearchResponseAdapter.js#L89

As a workaround, adding the field to the `include_fields` parameter prevents the exception. But I'd prefer to save the network bandwidth and not have unnecessary fields returned. Is this a bug or a design limitation?

**Comment by jasonbosco** [comment]
@themerk Could you share the network request and response to/from Typesense for this particular search query (from the browser's dev tools Network tab)? 

Copy-as-curl would be ideal for the request, and a JSON file with the response from Typesense would be ideal.

**Comment by themerk** [comment]
Here's the curl request:

```cmd
curl "https://78b3rjsu2f4hgzlyp-2.a1.typesense.net/multi_search?x-typesense-api-key=BtkZHVvfMHBX2d9QmTBx040Wh7zK9pKw" ^
  -H "authority: 78b3rjsu2f4hgzlyp-2.a1.typesense.net" ^
  -H "pragma: no-cache" ^
  -H "cache-control: no-cache" ^
  -H "sec-ch-ua: ^\^" Not A;Brand^\^";v=^\^"99^\^", ^\^"Chromium^\^";v=^\^"99^\^", ^\^"Google Chrome^\^";v=^\^"99^\^"" ^
  -H "accept: application/json, text/plain, */*" ^
  -H "content-type: text/plain" ^
  -H "sec-ch-ua-mobile: ?0" ^
  -H "user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36" ^
  -H "sec-ch-ua-platform: ^\^"Windows^\^"" ^
  -H "origin: https://www.echidnasewing.com.au" ^
  -H "sec-fetch-site: cross-site" ^
  -H "sec-fetch-mode: cors" ^
  -H "sec-fetch-dest: empty" ^
  -H "referer: https://www.echidnasewing.com.au/" ^
  -H "accept-language: en-US,en;q=0.9" ^
  --data-raw "^{^\^"searches^\^":^[^{^\^"highlight_fields^\^":^\^"^\^",^\^"query_by^\^":^\^"name,keywords,description,path_lvl_3,path_lvl_2,path_lvl_1,path_lvl_0^\^",^\^"highlight_full_fields^\^":^\^"name,path_lvl_3,path_lvl_2,path_lvl_1,path_lvl_0^\^",^\^"num_typos^\^":^\^"3^\^",^\^"include_fields^\^":^\^"type_code,name,url,img_url,path_lvl_3,path_lvl_2,path_lvl_1,path_lvl_0^\^",^\^"group_by^\^":^\^"type_code^\^",^\^"group_limit^\^":6,^\^"collection^\^":^\^"echidnasewing_content^\^",^\^"q^\^":^\^"debugkeyword^\^",^\^"facet_by^\^":^\^"visible_groups,delete^\^",^\^"filter_by^\^":^\^"visible_groups:=^[1^] && delete:=^[false^]^\^",^\^"page^\^":1^}^]^}" ^
  --compressed
```

Here's the payload of that request a little clearer:
```json
[
    {
        "highlight_fields": "",
        "query_by": "name,keywords,description,path_lvl_3,path_lvl_2,path_lvl_1,path_lvl_0",
        "highlight_full_fields": "name,path_lvl_3,path_lvl_2,path_lvl_1,path_lvl_0",
        "num_typos": "3",
        "include_fields": "type_code,name,url,img_url,path_lvl_3,path_lvl_2,path_lvl_1,path_lvl_0",
        "group_by": "type_code",
        "group_limit": 6,
        "collection": "echidnasewing_content",
        "q": "debugkeyword",
        "facet_by": "visible_groups,delete",
        "filter_by": "visible_groups:=[1] && delete:=[false]",
        "page": 1
    }
]
```

And here's the response:
```json
[
    {
        "facet_counts": [
            {
                "counts": [
                    {
                        "count": 1,
                        "highlighted": "1",
                        "value": "1"
                    },
                    {
                        "count": 1,
                        "highlighted": "2",
                        "value": "2"
                    },
                    {
                        "count": 1,
                        "highlighted": "3",
                        "value": "3"
                    },
                    {
                        "count": 1,
                        "highlighted": "4",
                        "value": "4"
                    },
                    {
                        "count": 1,
                        "highlighted": "5",
                        "value": "5"
                    }
                ],
                "field_name": "visible_groups",
                "stats": {
                    "avg": 3,
                    "max": 5,
                    "min": 1,
                    "sum": 15,
                    "total_values": 5
                }
            },
            {
                "counts": [
                    {
                        "count": 1,
                        "highlighted": "false",
                        "value": "false"
                    }
                ],
                "field_name": "delete",
                "stats": {
                    "total_values": 1
                }
            }
        ],
        "found": 1,
        "grouped_hits": [
            {
                "group_key": [
                    "category"
                ],
                "hits": [
                    {
                        "document": {
                            "img_url": "",
                            "name": "Christmas Designs",
                            "path_lvl_0": "Embroidery Designs",
                            "type_code": "category",
                            "url": "/embroidery-designs/christmas-designs/"
                        },
                        "highlights": [
                            {
                                "field": "keywords",
                                "indices": [
                                    3
                                ],
                                "matched_tokens": [
                                    [
                                        "debugkeyword"
                                    ]
                                ],
                                "snippets": [
                                    "<mark>debugkeyword</mark>"
                                ]
                            }
                        ],
                        "text_match": 3384305363909986
                    }
                ]
            }
        ],
        "out_of": 400,
        "page": 1,
        "request_params": {
            "collection_name": "echidnasewing_content_20220316",
            "per_page": 10,
            "q": "debugkeyword"
        },
        "search_cutoff": false,
        "search_time_ms": 2
    }
]
```

## Few things to note:
- The problem in this case is the `keywords` field (typed `string[]`). It is in the `query_by` param, but not the `include_fields`.
- My original report is wrong. It's a bit tricky to follow the SearchResponseAdapter code, but I think that exception is triggered because there is a highlight for the keywords field, but it's not actually present in the returned document.
- This is weird in itself, because I specifically request Typesense *not* to highlight `keywords` (`highlight_fields` is blank, and the field isn't listed in `highlight_full_fields`). Am I not setting the highlight params correctly?

**Comment by jasonbosco** [comment]
Fixed as of v0.23.0 of Typesense Server
